src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/TriggerHelperBindingProvider.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+    // Regular BindToInput has to do a TAttribute --> Value creation. 
+    // But triggers already have a Listener that provided the initial object; and we're just
+    // converting it ot the user's target parameter. 
+    internal class TriggerHelperBindingProvider<TAttribute, TTriggerValue> :
+        FluentBindingProvider<TAttribute>, IBindingProvider, IBindingRuleProvider
+           where TAttribute : Attribute
+            where TTriggerValue : class
+    {
+        private readonly INameResolver _nameResolver;
+        private readonly IConverterManager _converterManager;
+
+        public TriggerHelperBindingProvider(
+          INameResolver nameResolver,
+          IConverterManager converterManager
+          )
+        {
+            this._nameResolver = nameResolver;
+            this._converterManager = converterManager;
+        }
+
+        public Type GetDefaultType(Attribute attribute, FileAccess access, Type requestedType)
+        {
+            return typeof(object);
+        }
+
+        public IEnumerable<BindingRule> GetRules()
+        {
+            yield break;
```

Comment:
probably more efficient to return Enumerable.Empty<BindingRule>()? Not sure if the compiler does an optimization on this.